### PR TITLE
feat: Add new team editor - button and modal only

### PR DESCRIPTION
**Pull request by jamdelion**
**What is the feature?**
Add user in editor: https://trello.com/c/piMcGsJa/2973-add-user-in-editor

**What does this PR add?**
These changes are UI only!
- adds a button onto the members table that opens a modal. 
- The modal does nothing at the moment - both buttons just close the modal.
- The button is hidden behind a feature flag: 'ADD_NEW_EDITOR'

**Screenshots:**

Button:
Screenshot 2024-08-07 at 11 43 24

Modal:
Screenshot 2024-08-07 at 11 43 40

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/components/Team/MembersTable.tsx
Comment (jessicamcinchak) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/MembersTable.tsx:90]: I know this is still draft (looks like great start!) - but just wanted to flag that it might be useful to checkout `src/lib/featureFlags.ts` here - a small resource to help facilitate iterative PRs like this one :slightly_smiling_face: 

You could: 
- Add a new value to `AVAILABLE_FEATURE_FLAGS` array
- Add a condition to or replace `showAddMemberButton` with `hasFeatureFlag("{your new value}")`
- Run `window.featureFlags.toggle("{your new value")` in the browser console to turn on/off

Comment (jessicamcinchak) [reply editor.planx.uk/src/pages/FlowEditor/components/Team/MembersTable.tsx:90]: & an old example of how to then reference/use `toggleFeatureFlag` in tests ! https://github.com/theopensystemslab/planx-new/pull/1563/files

File: package.json
Comment (jamdelion) [inline package.json:18]: My VSCode version of typescript was causing some issues with VSCode refactoring tools, so I added this at the root level to allow me to switch to a lower version of TS. 

Sources: https://github.com/microsoft/TypeScript/issues/58897#issuecomment-2220808697 and https://github.com/yarnpkg/berry/issues/4788#issuecomment-1767059657

If anyone knows of a better way to get VSCode to pick up a new Typescript version, let me know! I tried `pnpm install -g`...

File: editor.planx.uk/src/lib/featureFlags.ts
Comment (jessicamcinchak) [inline editor.planx.uk/src/lib/featureFlags.ts:2]: nit: is this addition of `export` actually necessary / being used ?

File: editor.planx.uk/src/ui/editor/SettingsSection.tsx
Comment (jessicamcinchak) [inline editor.planx.uk/src/ui/editor/SettingsSection.tsx:35]: Handy addition, thanks :heavy_plus_sign:

File: package.json
Comment (jessicamcinchak) [reply package.json:18]: Strange one, haven't hit this before but happy to think on it a bit (shouldn't block merging this!). 

One thing I don't quite understand here though - why set version `5.4.5` here? I think I'd expect this to be `5.4.3` which is our lowest version of TS specified throughout this repo? (eg editor, scripts, e2e/tests/api-driven all on `5.4.3`; api `5.5.2`; and e2e root `5.5.3`?)

**Review by jessicamcinchak — APPROVED** [review]
Looks good! A couple minor questions, no blockers to merging. 

Thanks for clear to-do's and tests, exciting to see this feature taking shape :raised_hands:

File: editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx:2]: Wherever possible, we're trying to use the `setup()` API for consistency. See an example here -

https://github.com/theopensystemslab/planx-new/blob/main/editor.planx.uk/src/@planx/components/AddressInput/Public.test.tsx#L10-L26

Docs: 
- https://testing-library.com/docs/user-event/setup/
- https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
Fantastic PR - super clear and easy to follow, great level of explanation in the PR description and nice small atomic changes - love it 👍 

One very small comment on tests to take a look at please!

File: package.json
Comment (jamdelion) [reply package.json:18]: Good catch, I blithely followed the issue comments without thinking about what version would actually be appropriate! 😶‍🌫️ Switched to the lowest version now.

File: editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx
Comment (jamdelion) [reply editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx]: Ah ok, thanks for explaining! 

I've updated to be able to get `user` from `setup` (and thus got rid of `fireEvent` but is there a particular reason why we prefer use `getBy...` functions from here rather than from `screen`?

I had a go at getting everything from `setup` but in my opinion it made the tests a lot more unwieldy due to having to initialise the variables globally (`rendered`) and not being able to destructure them so nicely. 

I also think it made it make less sense to have a helper called `setupTeamMembersScreen` obscuring some of the methods that were exposed. I ended up with this which I don't think is as nice:

```js
describe("when a user with the ADD_NEW_EDITOR feature flag enabled presses 'add a new editor'", () => {
  let rendered: Record<"user", UserEvent> & RenderResult

  beforeEach(async () => {
    rendered = setup(
      <DndProvider backend={HTML5Backend}>
        <TeamMembers teamMembersByRole={exampleTeamMembersData} />
      </DndProvider>
    );
    await rendered.findByText("Team editors"); 
    const teamEditorsTable = rendered.getByTestId("team-editors");
    const addEditorButton = await within(teamEditorsTable).findByText(
      "Add a new editor"
    );
    rendered.user.click(addEditorButton);
  });
  it("opens the modal and displays the input fields", async () => {
    expect(await rendered.findByTestId("modal-create-user-button")).toBeVisible();
    expect(await rendered.findByLabelText("First name")).toBeVisible();
  });
});
```

In the example you linked, `screen` is used as well as `user` so hopefully the way it is left now is acceptable 😅  Super interested in talking through the thought processes behind the current code conventions though so let me know if you'd like a chat!

**Comment by jessicamcinchak** [comment]
:white_check_mark:  Tests are passing here & I don't want syntax/setup chat to block other dependent PRs, so I've "dismissed the requested changes" from Daf here just so you aren't blocked from merging this one for a couple days until he's back ! 

Let's pick back up the testing conversation later this week and can always come back for a light refactor for consistency if we decide :ok_hand:

**Pull request merged**
